sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayAsyncClient.java

```diff
@@ -36,62 +37,80 @@
+    }
+
+    /**
+     * Gets a Relay Configuration for a CommunicationUserIdentifier.
      *
      * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
-     * @return The created Communication Relay Configuration.
+     * @return The obtained Communication Relay Configuration.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<CommunicationRelayConfiguration> getRelayConfiguration(CommunicationUserIdentifier communicationUser) {
-        return this.getRelayConfigurationWithResponse(communicationUser).flatMap(FluxUtil::toMono);
+        return this.getRelayConfigurationWithResponse(communicationUser, null, null).flatMap(FluxUtil::toMono);
     }
 
     /**
-     * Creates a new CommunicationRelayConfiguration.
+     * Gets a Relay Configuration given a RouteType.
      *
-     * @return The created Communication Relay Configuration.
+     * @param routeType The specified RouteType for the relay request
+     * @return The obtained Communication Relay Configuration.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<CommunicationRelayConfiguration> getRelayConfiguration() {
-        return this.getRelayConfigurationWithResponse(null).flatMap(FluxUtil::toMono);
+    public Mono<CommunicationRelayConfiguration> getRelayConfiguration(RouteType routeType) {
+        return this.getRelayConfigurationWithResponse(null, routeType, null).flatMap(FluxUtil::toMono);
     }
 
     /**
-     * Creates a new CommunicationRelayConfiguration with response.
+     * Gets a Relay Configuration for a CommunicationUserIdentifier given a RouteType.
      *
      * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
-     * @return The created Communication Relay Configuration.
+     * @param routeType The specified RouteType for the relay request
+     * @return The obtained Communication Relay Configuration.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser) {
-        return withContext(context -> getRelayConfigurationWithResponse(communicationUser, context));
+    public Mono<CommunicationRelayConfiguration> getRelayConfiguration(CommunicationUserIdentifier communicationUser, RouteType routeType) {
+        return this.getRelayConfigurationWithResponse(communicationUser, routeType, null).flatMap(FluxUtil::toMono);
     }
 
     /**
-     * Creates a new CommunicationRelayConfiguration with response.
+     * Gets a Relay Configuration with response.
      *
-     * @return The created Communication Relay Configuration.
+     * @return The obtained Communication Relay Configuration.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse() {
```

Comment:
We should just have one `withReponse` right? Is there a reason why there are two here?

Reply:
Yes, the other one is not public since "@ServiceMethod must not has 'Context' as a method parameter". And that's the method all of the public ones are called, including the WithResponse that is public.